bip-0322.mediawiki

```diff
@@ -13,119 +13,99 @@
 
-{|class="wikitable" style="text-align: center;"
-|-
-!Code
-!Description
-|-
-|INCOMPLETE||Empty proof.
-|-
-|INCONCLUSIVE||The given proof was consensus-valid but policy-invalid.
-|-
-|VALID||The proof was valid.
-|-
-|INVALID||The proof was invalid
-|-
-|ERROR||An error was encountered
-|}
+where message_hash is a BIP340-tagged hash of the message, i.e. sha256_tag(m), where tag = "BIP0322-signed-message", and message_challenge is the to be proven (public) key script.
+For proving funds, message_challenge shall be simply OP_TRUE.
 
-== Signing and Verifying ==
+The "to_sign" transaction is:
 
-If the challenge consists of an address is in the P2PKH (legacy) format, sign using the legacy format (further information below). Otherwise continue as stated below.
+    nVersion = 0 or as appropriate (e.g. 2, for time locks)
+    nLockTime = 0 or as appropriate (for time locks)
+    vin[0].prevout.hash = to_spend.txid
+    vin[0].prevout.n = 0
+    vin[0].nSequence = 0 or as appropriate (for time locks)
+    vin[0].scriptWitness = message_signature
+    vout[0].nValue = 0
+    vout[0].scriptPubKey = OP_RETURN
 
-For both cases, generate a sighash based on the given scriptPubKey and message as follows:
+When a proof of funds is being created, additional inputs should be included for virtually spending transaction outputs of desired value.
 
-# Define the message pre-image as the sequence "Bitcoin Signed Message:\n" concatenated with the message, encoded in UTF-8 using Normalization Form Compatibility Decomposition (NFKD)
-# Let sighash = sha256(sha256(scriptPubKey || pre-image))
+* All signatures must use the SIGHASH_ALL flag.
+* The proof is considered valid, inconclusive, or invalid based on whether the to_sign transaction is a valid spend of the to_spend transaction or not, according to the rules specified in the "Consensus and standard flags" section below.
+* It may (optionally) be encumbered with the in_future flag, according to the rules specified in the "Locktime and Sequence" section below, in which case we refer to the result in text form as "valid_in_future", "inconclusive_in_future", etc.
 
-A private key may be used directly to sign a message. In this case, its P2WPKH bech32 address shall be derived, and used as the input.
+Proofs with additional inputs are the base64-encoding of the to_spend and to_sign transactions concatenated in standard network serialisation, and proofs without additional inputs (simple proofs) are the base64-encoding of the to_sign script witness.
 
-=== Signing ===
+A validator must verify it is valid and meets the description of virtual transactions as specified above. See "Validation" below.
 
-The signature is generated as follows:
+=== Validation ===
 
-# Derive the private key privkey for the scriptPubKey; FAIL if not VALID
-# Generate and return a signature sig with privkey=privkey, sighash=sighash
-
-=== Verifying ===
+To validate a proof with additional inputs, the following steps must be taken:
```

Comment:
```suggestion
To validate a proof of funds, the following steps must be taken:
```
We should always use the same term to refer to the same concept.

Reply:
Good point, yeah